### ISSUE TITLE
The ps_tag_count table

### ISSUE THREAD

**Issue by musicpanda** [PrestaShop/PrestaShop#37205]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

The ps_tag_count table has been introduced in PS 1.6.1.0 and fully implemented in 1.6.1.2. The table registers how popular the different tags are. That can then be used to produce tag clouds with the most popular tags. However, the implementation is incorrect.

The table has two indexes:
 - the primary index covers id_group and id_tag
 - the id_group index covers four fields: id_group, id_lang, id_shop and counter
 
The weird thing is that the primary index unique is. According to logic that should be the other index. As a result differentiation by shop and language is impossible. 

It looks like this feature was added in 1.6.1.0 for some customer who wanted a group differentiation. And then in 1.6.1.2 the table was adapted for much more useful differentiation by shop and language but this was never implemented in code or tested.

### Expected behavior

support for shops and languages

### Steps to reproduce

Look at the code

### PrestaShop version(s) where the bug happened

1.6.1.2 - 8.2

### PHP version(s) where the bug happened

8.1

**Comment by Hlavtox** [comment]
@musicpanda You are right, if there already is a pair of `id_group`-`id_tag`, it's impossible to add another row in this table.

The primary key should probably be ``(`id_group`, `id_lang`, `id_shop`, `counter`)``, then it should work.

However, `Tag::getMainTags` is not used by the core anywhere, you want to use it in a module?

Ping @kpodemski

**Comment by florine2623** [comment]
Seen with @jolelievre
